lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -45,6 +50,29 @@
 log = logging.getLogger("edx.discussions")
 
 
+class DiscussionCourseViewType(CourseViewType):
+    """
+    A tab for the cs_comments_service forums.
+    """
+
+    name = 'discussion'
+    title = _('Discussion')
+    priority = None
+    view_name = 'django_comment_client.forum.views.forum_form_discussion'
+
+    @classmethod
+    def is_enabled(cls, course, user=None):
+        has_course_access = True
+        if user:
+            has_course_access = (has_access(user, 'staff', course) or
+                                 CourseEnrollment.is_enrolled(user, course.id))
+
+        if settings.FEATURES.get('CUSTOM_COURSES_EDX', False):
+            if get_current_ccx():
```

Comment:
It is unfortunate that CCX leaks out into the discussion code. This isn't a new problem, but has just been made clearer by this refactoring. I can't think of a good way to separate this out, but I figured I'd flag it in case anyone else has ideas.